lib/fdotypes.c

```diff
@@ -4515,13 +4518,13 @@ bool fdo_rvto2addr_entry_read(fdor_t *fdor, fdo_rvto2addr_entry_t *rvto2addr_ent
 		}
 	}
 
-	if (fdor_is_value_null(fdor)) {
+	if (fdor_is_value_null(fdor) || !fdor_string_length(fdor, &rvdns_length) || rvdns_length == 0) {
 		if (!fdor_next(fdor)) {
 			LOG(LOG_ERROR, "RVTO2AddrEntry: Failed to skip NULL RVDNS\n");
 			return false;
 		}
 	} else {
-		size_t rvdns_length = 0;
+		rvdns_length = 0;
```

Comment:
Looks like this check is redundant since its already added in the previous condition. If yes, can this be removed ?

Reply:
it is required to get the length of the DNS